gobblin-metrics/src/main/java/gobblin/metrics/GobblinMetrics.java

```diff
@@ -728,33 +680,43 @@ private void buildCustomMetricReporters(Properties properties) {
-
-        if (CustomCodahaleReporterFactory.class.isAssignableFrom(clazz)) {
-          CustomCodahaleReporterFactory customCodahaleReporterFactory =
-              ((CustomCodahaleReporterFactory) clazz.getConstructor().newInstance());
-          com.codahale.metrics.ScheduledReporter scheduledReporter = this.codahaleReportersCloser
-              .register(customCodahaleReporterFactory.newScheduledReporter(RootMetricContext.get(), properties));
-          LOGGER.info("Will start reporting metrics to " + reporterClass);
-          this.scheduledReporters.add(scheduledReporter);
-        } else if (CustomReporterFactory.class.isAssignableFrom(clazz)) {
-          CustomReporterFactory customReporterFactory = ((CustomReporterFactory) clazz.getConstructor().newInstance());
-          customReporterFactory.newScheduledReporter(properties);
-        } else {
-          throw new IllegalArgumentException("Class " + reporterClass +
-              " specified by key " + ConfigurationKeys.METRICS_CUSTOM_BUILDERS + " must implement: "
-              + CustomCodahaleReporterFactory.class + " or " + CustomReporterFactory.class);
-        }
-      } catch (ClassNotFoundException exception) {
-        LOGGER.warn(String.format("Failed to create metric reporter: requested CustomReporterFactory %s not found.",
-                reporterClass), exception);
-      } catch (NoSuchMethodException exception) {
-        LOGGER.warn(String.format("Failed to create metric reporter: requested CustomReporterFactory %s "
-            + "does not have parameterless constructor.", reporterClass), exception);
-      } catch (Exception exception) {
-        LOGGER.warn("Could not create metric reporter from builder " + reporterClass + ".", exception);
+      buildScheduledReporter(properties, reporterClass, Optional.<String>absent());
+    }
+  }
+
+  private void buildScheduledReporter(Properties properties, String reporterClass, Optional<String> reporterSink) {
+    try {
+      Class<?> clazz = Class.forName(reporterClass);
+
+      if (CustomCodahaleReporterFactory.class.isAssignableFrom(clazz)) {
+        CustomCodahaleReporterFactory customCodahaleReporterFactory =
+            ((CustomCodahaleReporterFactory) clazz.getConstructor().newInstance());
+        com.codahale.metrics.ScheduledReporter scheduledReporter = this.codahaleReportersCloser
+            .register(customCodahaleReporterFactory.newScheduledReporter(RootMetricContext.get(), properties));
+        String reporterSinkMsg = reporterSink.isPresent()?"to " + reporterSink.get():"";
+        LOGGER.info("Will start reporting metrics " + reporterSinkMsg + " using " + reporterClass);
+        this.codahaleScheduledReporters.add(scheduledReporter);
+
+      } else if (CustomReporterFactory.class.isAssignableFrom(clazz)) {
+        CustomReporterFactory customReporterFactory = ((CustomReporterFactory) clazz.getConstructor().newInstance());
+        customReporterFactory.newScheduledReporter(properties);
+
```

Comment:
nit: a log message for easier tracking that this custom reporter was initialized